File: Classwork/Classwork09/BinaryTreeNode.java
/**
 *  File name: BinaryTreeNode.java
 *  Purpose: Demonstrates a binary tree
 *  @author: B.J. Johnson
 *  @date: 2019-10-23
 */

public class BinaryTreeNode {

   public  int data;
   private BinaryTreeNode left;
   private BinaryTreeNode right;

   BinaryTreeNode( int d ) {
       data = d;
       left = null;
       right = null;
   }

   public void add( int s, String child ) {
      child = child.toUpperCase();
      if( child.equals( "L" ) ) {
         left = new BinaryTreeNode( s );
      } else if( child.equals( "R" ) ) {
         right = new BinaryTreeNode( s );
      } else {
         throw new IllegalArgumentException();
      }
   }

   public BinaryTreeNode getChild( String child ) {
      child = child.toUpperCase();
      if( child.equals("L") || child.equals("R") ) {
         return (child.equals("L") ? left : right);
      } else {
         throw new IllegalArgumentException();
      }
   }

   public int getData() {
       return data;
   }
}

File: Classwork/classwork10/BinaryTreeNode.py
###
# filename: BinaryTreeNode.py
# purpose: binary tree demonstrator
# author:   Dr. Johnson
# date:     2023-02-23
###

class BinaryTreeNode:

   # the constructor
   def __init__( self, value ):
      self.data  = value
      self.left  = None
      self.right = None

   # add a child to the node
   def add_child( self, value, child ):
      child = child.upper()
      if( child == "L"  ):
         self.left = BinaryTreeNode( value )
      elif( child == "R" ):
         self.right = BinaryTreeNode( value )
      else:
         raise( "   Illegal Argument Exception..." )

   # get one child from this parent node
   def get_child( self, child ):
      child = child.upper()
      if( child == "L" ):
         return self.left
      elif( child == "R" ):
         return self.right
      else:
         raise( "   Illegal Argument Exception..." )

   # get this node's data value
   def get_data( self ):
       return self.data

